### DISCUSSION TITLE
How to use external command in mapping

### DISCUSSION THREAD

**Discussion by sarmong** [kovidgoyal/kitty#5755] [Q&A] [answered]
I want to create a mapping that calls and external script.

1. How to call the script from a keybind?
2. How to pipe output of `get-text --self` into external script on keybind?

**Comment by page-down** [comment, marked as answer]
> How to call the script ...

If you want kitty to run an external executable directly, such as opening it in a new window, use `launch` .
https://sw.kovidgoyal.net/kitty/launch/

If you want to run under the current shell in the current window and return to the shell after exiting, bind shortcuts in the shell, or use `send text`.
https://sw.kovidgoyal.net/kitty/conf/#shortcut-kitty.Send-arbitrary-text-on-key-presses

If you want to run custom Python scripts at the terminal level, use `kitten`.
https://sw.kovidgoyal.net/kitty/kittens/custom/

> How to pipe output of get-text ...

Use `launch --stdin-source` or use the remote control facility.
https://sw.kovidgoyal.net/kitty/launch/#the-piping-environment
https://sw.kovidgoyal.net/kitty/remote-control/#kitty-get-text
